rir/src/compiler/translations/pir_2_rir/pir_2_rir.cpp

```diff
@@ -919,6 +919,29 @@ rir::Code* Pir2Rir::compileCode(Context& ctx, Code* code) {
                       needsEnsureNamed.count(instr)))
                 cb.add(BC::ensureNamed());
 
+            // Check the return type
+            if (pir::Parameter::RIR_CHECK_PIR_TYPES > 0 &&
+                instr->type != PirType::voyd() &&
+                instr->type != NativeType::context && !CastType::Cast(instr) &&
+                Visitor::check(code->entry, [&](Instruction* i) {
+                    if (auto cast = CastType::Cast(i)) {
+                        if (cast->arg<0>().val() == instr)
+                            return false;
+                    }
+                    return true;
+                })) {
+                const char* instrStr;
+                if (pir::Parameter::RIR_CHECK_PIR_TYPES > 1) {
+                    std::stringstream instrPrint;
+                    instr->print(instrPrint, false);
+                    // WARNING: Leaks memory
+                    instrStr = (const char*)strdup(instrPrint.str().c_str());
```

Comment:
let's not do that. you can use an R string here and put it into the constant pool.

```
SEXP msg = Rf_mkString(debugMessage.c_str());
unsigned idx = Pool::insert(msg);
```

Reply:
then store that pool index in the bytecode stream